### DISCUSSION TITLE
modeling a multiplayer backend orchestrating many different types of actors

### DISCUSSION THREAD

**Discussion by mingqianye** [statelyai/xstate#3755] [Q&A]
I've been thinking about using xstate to implement a nodejs backend for a multiplayer boardgame-like platform. 

For simplicity sake, It has these entities:

1. `Users` (logged-in, logged-out, in-game, idle)
2. `Rooms` (active, archived, full, empty)
3. `Games` (staging, phase1, phase2, ..., ended)

My current design:
1. All those entities will be actors.
2. There's a `root` state machine. The sole purpose of the `root` state machine is to spawn these actors, and route events to one of these actors.
3. When the user sends an API request, the `root` machine forwards it to one of the actors.
4. The actors will communicate via their parent, i.e. the `root` state machine. For example, a `User` state machine can send event to the `root` machine, and the `root` machine can forward this event to one of the `Room` actors.
5. The `root` machine will just have a single state. It'll just have a very long `choose` action to route events to different actors and a self-transition.

Is it a good idea to use xstate like this, or am I missing something obvious here?

**Comment by davidkpiano** [comment]
Sorry for the delayed answer; it seems like XState would be a good idea for this, since you're modeling entities as actors (actor model) and their behavior can be represented as state machines.
